prime-dotnet-webapi/Services/EnrolleePaperSubmissionService.cs

```diff
@@ -220,6 +220,24 @@ public async Task UpdateSelfDeclarationsAsync(int enrolleeId, IEnumerable<PaperE
             var enrollee = await _context.Enrollees.Where(e => e.Id == enrolleeId).FirstOrDefaultAsync();
             enrollee.SelfDeclarationCompletedDate = DateTimeOffset.Now;
 
+            if (newDeclarations.Any())
+            {
+                var versions = await _context.Set<SelfDeclarationType>()
+                    .AsNoTracking()
+                    .Select(t => _context.Set<SelfDeclarationVersion>()
+                        .Where(av => av.EffectiveDate <= enrollee.SelfDeclarationCompletedDate)
+                        .Where(av => av.SelfDeclarationTypeCode == t.Code)
+                        .OrderByDescending(av => av.EffectiveDate)
+                        .First())
+                    .OrderBy(av => av.SelfDeclarationType.SortingNumber)
+                    .ToListAsync();
```

Comment:
This code seems to be in three other locations.  Can `LookupService,GetSelfDeclarationVersion` be used instead?

Reply:
As suggested, the code is updated to use LookupService to pull the versions in different locations.